src/main/java/io/strimzi/test/container/utils/KafkaVersionService.java

```diff
@@ -61,29 +50,37 @@ public static KafkaVersionService getInstance() {
     }
 
     /**
-     * Get complete image name from given Kafka and Strimzi image version
-     * If any of the given versions is {@code null} this method fetches the latest release versions using
-     * {@link KafkaVersionService}.
+     * Get whole image and if in {@link System#getProperties()} is specified field {@code strimzi.custom.image} then we use
+     * custom image. Moreover, if {@code kafkaVersion} is {@code null} this method fetches the latest release versions using
+     * {@link KafkaVersionService#getInstance()} instance.
      *
-     * @param strimziBaseImage Strimzi base image name
-     * @param strimziTestContainerImageVersion Strimzi test container image version
      * @param kafkaVersion Kafka version
-     * @return complete image name
+     * @return strimzi test container image path or custom image if Java property {@code strimzi.custom.image} is specified
      */
-    public static String strimziTestContainerImageName(String strimziBaseImage, String strimziTestContainerImageVersion, String kafkaVersion) {
-        if (strimziBaseImage == null || strimziBaseImage.isEmpty()) {
-            strimziBaseImage = STRIMZI_BASE_IMAGE;
-        }
-        if (strimziTestContainerImageVersion == null || strimziTestContainerImageVersion.isEmpty()) {
-            strimziTestContainerImageVersion = KafkaVersionService.getInstance().latestRelease().getStrimziTestContainerVersion();
-            LOGGER.info("No Strimzi test container version specified. Using latest release:{}", strimziTestContainerImageVersion);
-        }
-
-        if (kafkaVersion == null || kafkaVersion.isEmpty()) {
-            kafkaVersion = KafkaVersionService.getInstance().latestRelease().getVersion();
-            LOGGER.info("No Kafka version specified. Using latest release:{}", kafkaVersion);
+    public static String strimziTestContainerImageName(String kafkaVersion) {
```

Comment:
I'm not sure this really needs to be public? I know you have this `utils` package, which is why you need it to be public at the moment, but it's not clear to me why you really need a separate package. Having a single package would mean you could make this (and probably other methods) package access, which would reduce the API surface which you're exposing to uses, which in turn makes future refactoring much easier to reason about (they literally couldn't be calling internal methods like this one).

Reply:
So should I remove the `utils` package?